07.ls_object/ls_obj.rb

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+require './file_and_directory'
+require 'optparse'
+
+MAX_ROW = 3
+
+def main
+  opt = OptionParser.new
+
+  params = { dot_match: false, long_format: false, reverse: false }
+  opt.on('-a') { |v| params[:dot_match] = v }
+  opt.on('-l') { |v| params[:long_format] = v }
+  opt.on('-r') { |v| params[:reverse] = v }
+  opt.parse!(ARGV)
+  path = '.'
+
+  puts run_ls(path, **params)
+end
+
+def run_ls(path, dot_match: false, long_format: false, reverse: false)
+  filenames = collect_files(path, dot_match, reverse)
+  long_format ? list_long(filenames) : list_short(filenames)
+end
+
+def collect_files(path, dot_match, reverse)
+  filenames = dot_match ? Dir.foreach(path).to_a : Dir.foreach(path).filter { |file| !file.start_with?('.') }
```

Comment:
```suggestion
  filenames = Dir.foreach(path)
  filenames = dot_match ? filenames : filenames.reject { |file| file.start_with?('.') }
```

みたいにリファクタリングできそうです。
ただし、変数の再代入はなるべく避けた方がいいので、変数の命名をちょっと検討してみてください。
https://bootcamp.fjord.jp/pages/avoid-re-assign

Reply:
filesという変数に置きました